src/Storages/StorageFile.cpp

```diff
@@ -164,6 +165,12 @@ bool StorageFile::isColumnOriented() const
 StorageFile::StorageFile(int table_fd_, CommonArguments args)
     : StorageFile(args)
 {
+    struct stat buf;
+    int res = fstat(table_fd_, &buf);
+    if (-1 == res)
+        throwFromErrno("Cannot execute fstat", res, ErrorCodes::CANNOT_FSTAT);
+    total_bytes_to_read = buf.st_size;
```

Comment:
Ok. What happens if fd is a pipe?

Reply:
Zero bytes, no exception, no progress bar. But it is expected behaviour I think.